RedHat-UX/red-hat-design-system
# feat(cta): add rh-cta

## What I did

1. implement rh-cta
2. add color context to `lib` (will eventually be removed from pfe, so import from here from now on)
3. add `DirController` - useful when you need to imperatively style things based on rtl

### Testing Instructions
- Examine each interaction state (default, hover, focus, active) for each CTA in https://deploy-preview-349--red-hat-design-system.netlify.app/components/cta/demo/
- `npm ci && npm run test:e2e`. Inspect the test files generated in `element/rh-cta/test/`
- @heyMP please check the integration with footer
- @zeroedin please check the integration with secondary nav. We might want to migrate secondary nav to use the local context decorators, but if this PR works, we can do that later.
- we're missing the video icon, so i didn't implement it. see discussion

## Notes to reviewers
- Please carefully review the token names used in the variants section of rh-cta.css - do all the token names used represent their semantic use? i.e. did I use border color tokens for instances of border?
- The brick-on-dark styles were not specified, so I improvised
- there is no `on saturated` as it wasn't specified
### Old Notes
This depends a lot on design-tokens, and there are lots of unanswered questions there at the time of this writing.
The biggest single feature of CTA is theming and context (which also includes `broadcast`), and it's not clear what the future holds for those concepts in both pfe and rhds. Please see the relevant discussion:
- https://github.com/RedHat-UX/red-hat-design-tokens/pull/2
- https://github.com/orgs/RedHat-UX/discussions/348
- https://github.com/orgs/RedHat-UX/discussions/306
Ensure this fixes https://github.com/RedHat-UX/red-hat-design-system/issues/238


Closes #333
